pandas/core/strings.py

```diff
@@ -817,12 +817,15 @@ def _str_extract_frame(arr, pat, flags=0):
         result_index = arr.index
     except AttributeError:
         result_index = None
-    return DataFrame(
+    result = DataFrame(
         [groups_or_na(val) for val in arr],
         columns=columns,
         index=result_index,
         dtype=object,
```

Comment:
can you directly specify the dtype here? (instead of the below) or is that path not working?